xbmc/addons/AddonManager.cpp

```diff
@@ -479,7 +449,61 @@ bool CAddonMgr::HasAvailableUpdates()
   return !GetAvailableUpdates().empty();
 }
 
-bool CAddonMgr::GetAddons(const TYPE &type, VECADDONS &addons, bool enabled /* = true */)
+bool CAddonMgr::GetAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, true);
+}
+
+bool CAddonMgr::GetAddons(VECADDONS& addons, const TYPE& type)
+{
+  return GetAddonsInternal(type, addons, true);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, false);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons, const TYPE& type)
```

Comment:
The whole goal of the PR seems to be to get rid of the optional parameters because they are non-obvious when reading calls. You just see a `true` or a `false` but you have no clue what it means. Overloads with specific names make it more obvious to figure out what's happening.